llvm/lib/Target/TargetMachine.cpp

```diff
@@ -55,17 +52,22 @@ bool TargetMachine::isLargeGlobalObject(const GlobalObject *GO) const {
     return false;
 
   // Allowing large metadata sections in the presence of an explicit section is
-  // useful, even if GCC does not allow them. However, we should not mark
-  // certain well-known prefixes as large, because it would make the whole
-  // output section large and cause the linker to move it, which is almost
-  // always undesired.
+  // useful, even if GCC does not allow them. However, we should properly mark
```

Comment:
The first sentence of this comment doesn't make sense anymore. You can probably delete everything up to and including "However,". Regardless of the code model, we try to infer the large-ness of the global from certain well-known explicit section names.

This is a micro-nit / premature optimization, but I'd throw all this code into a `if (!Name.empty()) {` conditional, as a way to mean "if we have an explicit section, here's a bunch of logic you can skip over".